org.elasticsearch.add_retention_lease.RetentionLeaseBwcIT.testRetentionLeaseBwcIT Test failure

testRetentionLeaseBwcIT has caused a handful of 6.7 BWC builds to fail.
Failure reproduces locally
Reproduce lines: 
```
./gradlew :qa:retention-lease-bwc:v6.7.0#newClusterTestRunner -Dtests.seed=7617A5AA63CC7356 -Dtests.class=org.elasticsearch.add_retention_lease.RetentionLeaseBwcIT -Dtests.method="testRetentionLeaseBwcIT" -Dtests.security.manager=true -Dtests.locale=pl -Dtests.timezone=America/Grand_Turk -Dtests.distribution=zip -Dcompiler.java=12 -Druntime.java=8
```

Failure stack trace:
```
4:13:11    > Throwable #1: java.net.SocketTimeoutException
14:13:11    > 	at __randomizedtesting.SeedInfo.seed([7617A5AA63CC7356:BFE88988248466E5]:0)
14:13:11    > 	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:944)
14:13:11    > 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:233)
14:13:11    > 	at org.elasticsearch.test.rest.ESRestTestCase.ensureGreen(ESRestTestCase.java:786)
14:13:11    > 	at org.elasticsearch.add_retention_lease.RetentionLeaseBwcIT.testRetentionLeaseBwcIT(RetentionLeaseBwcIT.java:81)
14:13:11    > 	at java.lang.Thread.run(Thread.java:748)
14:13:11    > Caused by: java.net.SocketTimeoutException
14:13:11    > 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:375)
14:13:11    > 	at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:92)
14:13:11    > 	at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:39)
14:13:11    > 	at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
14:13:11    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:263)
14:13:11    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:492)
14:13:11    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:213)
14:13:11    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
14:13:11    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
14:13:11    > 	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
14:13:11    > 	... 1 more
14:13:11 Completed [1/1] in 34.59s, 1 test, 1 error <<< FAILURES!
```

Running locally, there is nothing in the node logs (new and old) and indicate a start up failure. The new node spins up just fine, and is added to the cluster. 

Investigating further.

--- a/qa/retention-lease-bwc/build.gradle
+++ b/qa/retention-lease-bwc/build.gradle
@@ -30,13 +30,17 @@ esplugin {
 integTest.enabled = false
 
 task bwcTest {
-  description = 'runs retention lease backwards compatability tests'
+  description = 'runs retention lease backwards compatibility tests'
   group = 'verification'
 }
 
 for (Version version : bwcVersions.wireCompatible) {
   if (version.before("6.5.0")) {
     // versions before 6.5.0 do not support soft deletes
+    continue
+  }
+  if (version.onOrAfter("6.7.0")) {
+    // we are known to be compatible with these versions
     continue
   }
 

